src/common/gitProvider/bitbucket.ts

```diff
@@ -0,0 +1,226 @@
+import { GitProviderRoot } from "./gitProviderRoot";
+import * as c from "chalk";
+import { PullRequestMessageRequest, PullRequestMessageResult } from ".";
+import { git, uxLog } from "../utils";
+import { Bitbucket, Schema } from "bitbucket";
+
+export class BitbucketProvider extends GitProviderRoot {
+    private bitbucket: InstanceType<typeof Bitbucket>;
+
+    constructor(){
+        super();
+        const token = process.env.CI_SFDX_HARDIS_BITBUCKET_TOKEN;
+        const clientOptions = { auth: { token: token }};
+        this.bitbucket = new Bitbucket(clientOptions);
+    }
+
+    public getLabel(): string {
+        return "sfdx-hardis Bitbucket connector";
+    }
+
+    public async getCurrentJobUrl(): Promise<string> {
+        if (process.env.BITBUCKET_WORKSPACE && process.env.BITBUCKET_REPO_SLUG && process.env.BITBUCKET_BUILD_NUMBER){
+            const jobUrl = `https://bitbucket.org/${process.env.BITBUCKET_WORKSPACE}/${process.env.BITBUCKET_REPO_SLUG}/pipelines/results/${process.env.BITBUCKET_BUILD_NUMBER}`;
+            return jobUrl;
+        }
+        uxLog(
+            this,
+            c.yellow(`[Bitbucket Integration] You need the following variables to be accessible to sfdx-hardis to build current job url:
+        - BITBUCKET_WORKSPACE
+        - BITBUCKET_REPO_SLUG
+        - BITBUCKET_BUILD_NUMBER`),
+          );
+
+        return null;
+    }
+
+    public async getCurrentBranchUrl(): Promise<string> {
+        if (
+            process.env.BITBUCKET_WORKSPACE &&
+            process.env.BITBUCKET_REPO_SLUG &&
+            process.env.BITBUCKET_BRANCH
+        ) {
+            const currentBranchUrl = `https://bitbucket.org/${process.env.BITBUCKET_WORKSPACE}/${process.env.BITBUCKET_REPO_SLUG}/branch/${process.env.BITBUCKET_BRANCH}`;
```

Comment:
Can the base url of bitbucket be taken from an ENV var ?
In case bitbucket is on premise, not on bitbucket.org :)

Reply:
I've looked into this and the Node.js API Client used for making the API calls to Bitbucket doesn't support on-premise servers, as Bitbucket Server doesn't include the latest versions of the REST API. 
https://github.com/MunifTanjim/node-bitbucket/issues/29

To add support for Bitbucket Server would probably require a separate GitProvider class creating, as the implementation would need to be quite different. It also looks like Atlassian ended support for Bitbucket Server a couple of months ago too, so I expect not many teams will use it going forwards.
https://www.atlassian.com/migration/assess/journey-to-cloud

Reply:
Good explanation, ok for me :)